tests/uart_tests.py

```diff
@@ -0,0 +1,149 @@
+import threading
+import unittest, sys, os
+from supernovacontroller.sequential import SupernovaDevice
+from BinhoSupernova.commands.definitions import UartControllerBaudRate as baud, UartControllerParity as parity, UartControllerDataSize as dataSize, UartControllerStopBit as stopBit
+
+from supernovacontroller.sequential.uart import SupernovaUARTBlockingInterface
+
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
+from binhosimulators import BinhoSupernovaSimulator
+
+# TODO @BMC2-1512 Update the simulator to support the execution/usage via SNController 
+class TestSupernovaControllerUART(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """
+        Initializes the testing class. Determines whether to use the simulator or real device
+        based on the "USE_REAL_DEVICE" environment variable. Default is to use the simulator.
+        """
+        cls.use_simulator = False
+
+    def setUp(self):
+        self.device = SupernovaDevice()
+        if self.use_simulator:
+            self.device.driver = BinhoSupernovaSimulator()
+
+    def test_uart_echo_message_real_device(self):
+        if self.use_simulator:
+            self.skipTest("For real device only")
+
+        self.device.open()
+        uart : SupernovaUARTBlockingInterface = self.device.create_interface("uart") # Type hinting for easier development
+
+        (success, _) = uart.set_bus_voltage(3300)
+        uart.init_bus()
+
+        data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+
+        (success, _) = uart.send(data)
+        (successReceive, response) = uart.wait_for_notification(1)
+
+        self.assertEqual(success, True)
+        self.assertTupleEqual((successReceive, response), (True, data), f"Failed echo: {response}")
+
+        self.device.close()
```

Comment:
Consider using [tearDown()](https://docs.python.org/3/library/unittest.html#unittest.TestCase.tearDown) to close the device in each test.

Reply:
This apply for all the tests.